run-clang-format.py

```diff
@@ -31,13 +31,26 @@
 
 
 DEFAULT_EXTENSIONS = 'c,h,C,H,cpp,hpp,cc,hh,c++,h++,cxx,hxx'
+DEFAULT_CLANG_FORMAT_IGNORE= '.clang-format-ignore'
 
 
 class ExitStatus:
     SUCCESS = 0
     DIFF = 1
     TROUBLE = 2
 
+def get_clang_format_ignore_excludes(clang_format_file):
```

Comment:
I would try to make the try/except stricter.
Only ignoring a "No such file or directory" error, on the `io.open()` call.
Any other errors should be reported to the user so that it can try to act on it accordingly.

This code seems ok to me, for Python 2 & 3 portability:
- https://stackoverflow.com/a/55755432/951426